docker-compose.yml

```diff
@@ -0,0 +1,821 @@
+      ASTONIA_DATABASE_HOSTNAME: ${DATABASE_HOSTNAME}
+      ASTONIA_USERNAME: ${DATABASE_USERNAME}
+      ASTONIA_DATABASE_PORT: ${DATABASE_PORT}
+    depends_on:
+      db:
+        condition: service_healthy
+    ports:
+      - "6688:6688"
+    command: ["-a 33", "-p6688", "-b${ASTONIA_3_SERVER_ADDRESS}"]
+    volumes:
+      - long_tunnel_zone_data:/opt/var/astonia/zones/33
+      - generic_zone_data:/opt/var/astonia/zones/generic
+      - preset_zone_data:/opt/var/astonia/zones/preset
+    configs:
+      - source: message_of_the_day
+        target: /opt/var/astonia/motd.txt
+  teufelheim:
+    build:
+      context: .
+    environment:
+      ASTONIA_DATABASE_HOSTNAME: ${DATABASE_HOSTNAME}
+      ASTONIA_USERNAME: ${DATABASE_USERNAME}
+      ASTONIA_DATABASE_PORT: ${DATABASE_PORT}
+    depends_on:
+      db:
+        condition: service_healthy
+    ports:
+      - "6689:6689"
+    command: ["-a 34", "-p6689", "-b${ASTONIA_3_SERVER_ADDRESS}"]
+    volumes:
+      - teufelheim_zone_data:/opt/var/astonia/zones/34
+      - generic_zone_data:/opt/var/astonia/zones/generic
+      - preset_zone_data:/opt/var/astonia/zones/preset
+    configs:
+      - source: message_of_the_day
+        target: /opt/var/astonia/motd.txt
+  caligar:
+    build:
+      context: .
+    environment:
+      ASTONIA_DATABASE_HOSTNAME: ${DATABASE_HOSTNAME}
+      ASTONIA_USERNAME: ${DATABASE_USERNAME}
+      ASTONIA_DATABASE_PORT: ${DATABASE_PORT}
+    depends_on:
+      db:
+        condition: service_healthy
+    ports:
+      - "6691:6691"
+    command: ["-a 36", "-p6691", "-b${ASTONIA_3_SERVER_ADDRESS}"]
+    volumes:
+      - caligar_zone_data:/opt/var/astonia/zones/36
+      - generic_zone_data:/opt/var/astonia/zones/generic
+      - preset_zone_data:/opt/var/astonia/zones/preset
+    configs:
+      - source: message_of_the_day
+        target: /opt/var/astonia/motd.txt
+  arkhata:
+    build:
+      context: .
+    environment:
+      ASTONIA_DATABASE_HOSTNAME: ${DATABASE_HOSTNAME}
+      ASTONIA_USERNAME: ${DATABASE_USERNAME}
+      ASTONIA_DATABASE_PORT: ${DATABASE_PORT}
+    depends_on:
+      db:
+        condition: service_healthy
+    ports:
+      - "6692:6692"
+    command: ["-a 37", "-p6692", "-b${ASTONIA_3_SERVER_ADDRESS}"]
+    volumes:
+      - arkhata_zone_data:/opt/var/astonia/zones/37
+      - generic_zone_data:/opt/var/astonia/zones/generic
+      - preset_zone_data:/opt/var/astonia/zones/preset
+    configs:
+      - source: message_of_the_day
+        target: /opt/var/astonia/motd.txt
+  chat:
+    build:
+      context: .
+    environment:
+      ASTONIA_DATABASE_HOSTNAME: ${DATABASE_HOSTNAME}
+    ports:
+      - "6654:6654"
+    expose:
+      - "6654"
+    entrypoint: ["/opt/var/astonia/chatserver"]
+
+configs:
+  message_of_the_day:
+    content: |
+      Using /punish or /shutup or /exterminate is strictly forbidden. You're here to test, not to fool around.
+
+secrets:
+  database_password:
+    environment: "DATABASE_PASSWORD"
```

Comment:
Since my branch was stale when I submitted the pull request, I didn't have a SQLPASSWORD file to work with, so I leveraged an environment variable that is seeded into the container ephemerally. Technically this could now point to the mysqlpassword file.